src/ua_types.c

```diff
@@ -1114,3 +1114,12 @@ UA_Array_delete(void *p, size_t size, const UA_DataType *type) {
     }
     UA_free((void*)((uintptr_t)p & ~(uintptr_t)UA_EMPTY_ARRAY_SENTINEL));
 }
+
+UA_Boolean
+isDataTypeNumeric(const UA_DataType *type) {
```

Comment:
There could also be custom subtypes which derive from the base datatypes. These should also be handled.

Reply:
Hi, can you please shortly explain how i can detect a type not already existing?

in the filtering code i do:
if (UA_Variant->type == &UA_TYPES[UA_TYPES_SBYTE])
((const UA_SByte*)UA_Variant->data)[index] to get the right value for array type variant.
Then I do: (a>b?a-b:b-a)
to compare with dead band value.

See outOfDeadBand in ua_subscription_datachange.c:154

How can I do that for custom types ?

Then I do
if(type == &UA_TYPES[UA_TYPES_SBYTE])
to detect if a type is a numeric type.

See isDataTypeNumeric in ua_types.c:1119

How can I do that for custom types ?

Reply:
@Pro Can we do that in an additional PR if the need comes up?
We then need an additional mechanism that ensures that the encoding is identical to that of the base datatype. Because we cast to double/float/int_x for the comparison.